core/src/main/java/com/linecorp/armeria/client/retry/RetryingClientBuilder.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Builds a new {@link RetryingClient} or its decorator function.
+ *
+ * @param <T> the type of {@link RetryingClientBuilder}
+ * @param <U> the type of the {@link Client} that this builder builds or decorates
+ * @param <I> the type of outgoing {@link Request} of the {@link Client}
+ * @param <O> the type of incoming {@link Response} of the {@link Client}
+ */
+public abstract class RetryingClientBuilder<
+         T extends RetryingClientBuilder<T, U, I, O>, U extends RetryingClient,
+         I extends Request, O extends Response> {
+
+    final RetryRequestStrategy<I, O> retryStrategy;
+    Supplier<? extends Backoff> backoffSupplier = Backoff::withoutDelay;
```

Comment:
I'm curious if this supplier is the sane default. How about changing this to something safer taking advantage of this change?